aws/resource_aws_cloudwatch_event_archive_test.go

```diff
@@ -0,0 +1,230 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	events "github.com/aws/aws-sdk-go/service/cloudwatchevents"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func init() {
+	resource.AddTestSweepers("aws_cloudwatch_event_archive", &resource.Sweeper{
+		Name: "aws_cloudwatch_event_archive",
+		F:    testSweepCloudWatchEventArchives,
+		Dependencies: []string{
+			"aws_cloudwatch_event_bus",
+		},
+	})
+}
+
+func testSweepCloudWatchEventArchives(region string) error {
+	client, err := sharedClientForRegion(region)
+	if err != nil {
+		return fmt.Errorf("Error getting client: %w", err)
+	}
+	conn := client.(*AWSClient).cloudwatcheventsconn
+
+	input := &events.ListArchivesInput{}
+
+	for {
+		output, err := conn.ListArchives(input)
+		if err != nil {
+			if testSweepSkipSweepError(err) {
+				log.Printf("[WARN] Skipping CloudWatch Events archive sweep for %s: %s", region, err)
+				return nil
+			}
+			return fmt.Errorf("Error retrieving CloudWatch Events archive: %w", err)
+		}
+
+		if len(output.Archives) == 0 {
+			log.Print("[DEBUG] No CloudWatch Events archives to sweep")
+			return nil
+		}
+
+		for _, archive := range output.Archives {
+			name := aws.StringValue(archive.ArchiveName)
+			if name == "default" {
+				continue
+			}
+
+			log.Printf("[INFO] Deleting CloudWatch Events archive (%s)", name)
+			_, err := conn.DeleteArchive(&events.DeleteArchiveInput{
+				ArchiveName: aws.String(name),
+			})
+			if err != nil {
+				return fmt.Errorf("Error deleting CloudWatch Events archive (%s): %w", name, err)
+			}
+		}
+
+		if output.NextToken == nil {
+			break
+		}
+		input.NextToken = output.NextToken
+	}
+
+	return nil
+}
+
+func TestAccAWSCloudWatchArchive_basic(t *testing.T) {
+	var v1 events.DescribeArchiveOutput
+	archiveName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_cloudwatch_event_archive.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSCloudWatchArchiveDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSCloudWatchArchiveConfig(archiveName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckCloudWatchArchiveExists(resourceName, &v1),
+					resource.TestCheckResourceAttr(resourceName, "name", archiveName),
```

Comment:
To make sure everything is working as expected, you should check all attributes are set correctly.